model_server_tools/model_packaging/export_model_utils.py

```diff
@@ -0,0 +1,289 @@
+        symbol_file = '%s-symbol.json' % model_name
+        params_file = '%s-0000.params' % model_name
+        signature_file = 'signature.json'
+        # Find input symbol name and shape
+        model_proto = onnx.load(os.path.join(model_path, onnx_file))
+        graph = model_proto.graph
+        _params = set()
+        for tensor_vals in graph.initializer:
+            _params.add(tensor_vals.name)
+
+        input_data = []
+        for graph_input in graph.input:
+            shape = []
+            if graph_input.name not in _params:
+                for val in graph_input.type.tensor_type.shape.dim:
+                    shape.append(val.dim_value)
+                input_data.append((graph_input.name, tuple(shape)))
+
+        sym, arg_params, aux_params = onnx_mxnet.import_model(os.path.join(model_path, onnx_file))
+        # UNION of argument and auxillary parameters
+        params = dict(arg_params, **aux_params)
+        # rewrite input data_name correctly
+        with open(os.path.join(model_path, signature_file), 'r') as f:
+            data = json.loads(f.read())
+            data['inputs'][0]['data_name'] = input_data[0][0]
+        with open(os.path.join(model_path, signature_file), 'w') as f:
+            f.write(json.dumps(data, indent=2))
+
+        with open(os.path.join(model_path, symbol_file), 'w') as f:
+            f.write(sym.tojson())
+
+        save_dict = {('arg:%s' % k): v.as_in_context(mx.cpu()) for k, v in params.items()}
+        mx.nd.save(os.path.join(model_path, params_file), save_dict)
+        return symbol_file, params_file
+
+    @staticmethod
+    def generate_publisher(publisherargs):
+        publisher = Publisher(author=publisherargs.author, email=publisherargs.email)
+        return publisher
+
+    @staticmethod
+    def generate_engine(engineargs):
+        engine = Engine(engine_name=engineargs.engine)
+        return engine
+
+    @staticmethod
+    def generate_model(modelargs):
+        model = Model(model_name=modelargs.model_name, handler=modelargs.handler)
+        return model
+
+    @staticmethod
+    def generate_manifest_json(args):
+        """
+        Function to generate manifest as a json string from the inputs provided by the user in the command line
+        :param args:
+        :return:
+        """
+
+        publisher = ModelExportUtils.generate_publisher(args) if args.author and args.email else None
+
+        engine = ModelExportUtils.generate_engine(args)
+
+        model = ModelExportUtils.generate_model(args)
+
+        manifest = Manifest(runtime=args.runtime, engine=engine, model=model, publisher=publisher)
+
+        return str(manifest)
+
+    @staticmethod
+    def create_manifest_file(model_path, manifest):
+        """
+        Function creates a file called manifest.json under model_path/MAR-INF folder
+
+        :param model_path:
+        :param manifest:
+        :return:
+        """
+        mar_inf_path = os.path.join(model_path, MAR_INF)
```

Comment:
We don't need create this folder. We can add MAR_INF entry directly in zip file.

Reply:
MAR_INF needs to created somewhere before being added to the Zip file. It cannot be directly added to zip without first creating it.

Reply:
Done in a46fa5d65e8493d56fda684f0cca7f0e97d7cf5c